D2k map pool update - november 2023

Porenutak opened the pull request:
First commit adds new cliff passages that these (and manny other) maps use. All new tiles was created by me and are free to use.
Second commit update map pool:
new maps: worm_graveyard.oramap
update maps: red-chasm.oramap, death-depths.oramap, the-duell.oramap
removed maps: cliffmaze

Updated maps changelog: equal distance from spice fields for each player, equal amount of spice for each player and minor balance adjustments.

Ps. Worm graveyard use repairble passages by enginear.

PunkPun commented:
I dunno why but this PR needs a rebase

Porenutak commented:
I rebase on my local branch, but when I try to pull it on github I get this error:
![Screenshot (243)](https://github.com/OpenRA/OpenRA/assets/100044015/93ee8dac-e2f8-4bf5-8fc2-81f814f732f5)
I also cant Squash rebase commits:
![Screenshot (245)](https://github.com/OpenRA/OpenRA/assets/100044015/c4937644-51c6-45a6-b3c1-061bd2adc8a3)

Can someone help me with this?
Im noob with Github and not sure whats wrong.

pchote reviewed (commented):
A few nits:
* worm_graveyard.oramap contains custom rules, so should be commited as an unpacked map.
* Please rename NewTiles.shp to something more descriptive.
* Please keep other filenames lower case (worm_graveyard's custom yaml uses a mix of CamelCase and lower case)

Porenutak commented:
Updated according to your suggestions.
NewTiles.shp renamed to custompassagetiles.shp

PunkPun commented on mods/d2k/maps/worm-graveyard/destroyable-tiles.yaml:
```diff
@@ -0,0 +1,198 @@
+		ValidRelationships: Enemy, Neutral, Ally
+
+
+
```
this triple whitespace is pretty weird, we don't do this. There's also some double whitespaces in this file

PunkPun reviewed (commented):
I'm not sure how to feel about this feature. It's very hidden away so likely only spreadable by mouth to mouth. It also feels unpolished, should tooltips be in lower case?

<img width="307" alt="Passage2" src="https://github.com/OpenRA/OpenRA/assets/37534529/b4d6f07e-46b2-4ff6-a224-f3c46f6eec4b">

<img width="519" alt="Passage1" src="https://github.com/OpenRA/OpenRA/assets/37534529/a0643e52-f433-4f91-b49f-5a9563c8f3c1">

https://github.com/OpenRA/OpenRA/assets/37534529/43b50955-6104-424e-bb8c-7ab665365a70

Porenutak commented:
This feature is here -/+ year and a half. Its part of Enhanced maps and other map mods inherit it to. The reason why engi didnt dissapier is because of balance. Noone wanna spend 400$ just for cliff repair that can be destroyed again. 
Ussually this feautre is used as obsticle to capture neutral building in early game.
![Snímka obrazovky (26)](https://github.com/OpenRA/OpenRA/assets/100044015/235ad994-ab8d-4d53-be40-c25ee5b1b1bd)
![Snímka obrazovky (25)](https://github.com/OpenRA/OpenRA/assets/100044015/cb4d9316-c395-47e1-b363-7a227bcbb4b1)

If you feel its a problem a can talk to map author if he agree to remove them. 
My English is terrible, so if someone come with better tooltip description I will include  it.

Porenutak commented:
> I'm not sure how to feel about this feature. It's very hidden away so likely only spreadable by mouth to mouth. It also feels unpolished, should tooltips be in lower case?>

Maybe I should add Briefing description?

PunkPun commented:
I'm not sure what's the best way.

Perhaps the fist step would be not requiring force attack to kill it? It just feels like the cliff is basically invisible and one has no way to learn about it unless explicitly taught

Porenutak commented:
This feature should act same as for example bridges in RA. You dont wanna autotarget brige, you wanna past thru it or stay there or destroy it only when situation require that (same as in RA). So force force is the good way. Repaireble cliffs are shown on Radar as actors and tooltip give you more info. I dont know how to hint players more without explicit description in mission briefing (which I do when this feautre come out)

Porenutak commented:
Rebased.

Worm cementery map:
- custom tiles converted from r8 - r16
- add description to engineer, that he can repair destroyed cliffs
- changed destroyed cliffs into repaired state, so even if someone didnt notice this map feature it wouldnt be a problem.

PunkPun commented on mods/d2k/maps/worm-graveyard/destroyable-tiles.yaml:
```diff
@@ -0,0 +1,198 @@
+		HP: 30000
+	Tooltip:
+		GenericName: pasage (Destroable)
+		Name: passage (Destroable)
```
```suggestion
		Name: Passage (destroyable)
```

these should be using our translations system, and do standardise the capital letters and fix spelling

Porenutak replied:
Do we have some dokumentation on how translations works? Can I just overide en.ftl in  custom maps?

PunkPun replied:
you can run the `--extract-yaml-strings` command. Then you can format it to match the others

PunkPun replied:
the system is currently in progress, and no, maps can't overwrite mod level translations. You need unique translation keys. The lint checks, that is `--check-yaml` should catch any errors you might have. Though the fact that this PR didn't upset the CI gods makes me suspicious that something broke

Porenutak commented:
Map rules use translation now.

PunkPun commented:
The discussion on discord indicates that there's a desire to have tiles defined at a game level instead of by the map

Porenutak commented:
- Destroyable tiles now defined inside the mod. Assets moved into bits folder
- engineer description updated
- fixed AI attacking cliffs

Porenutak commented:
removed some redundant lines in destroyable-tiles.yaml

PunkPun commented on mods/d2k/languages/rules/en.ftl:
```diff
@@ -96,8 +96,9 @@ actor-light-inf =
-      Weak vs Everything
+     Strong vs Buildings
+     Weak vs Everything
+     Can repair destroyed cliffs
```
the rest of the file has 3 space intendation, while this has 2

PunkPun commented on mods/d2k/languages/rules/en.ftl:
```diff
@@ -472,3 +473,12 @@ actor-combat-tank-o =
 
     The Ordos Combat Tank is the fastest variant of the Combat Tank, but it is also the weakest.
+
+
```
please remove double whitespace

PunkPun reviewed (dismissed):
LGTM

Mailaender committed:
Map pool update

- add Destroyable tiles into d2k mod
- add ability engineer to repair destroyed cliffs

Mailaender commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/6b1d8d2eae959705a0b482316dc8ca9230ac37a7...e4bbbcd838af1b38e2a22f5f4355a6e418390e04)
